core/services/relay/evm/chain_writer.go

```diff
@@ -0,0 +1,122 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+	commonservices "github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink/v2/common/txmgr"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config"
+	evmtxmgr "github.com/smartcontractkit/chainlink/v2/core/chains/evm/txmgr"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+type ChainWriterService interface {
+	services.ServiceCtx
+	types.ChainWriter
+}
+
+// Compile-time assertion that chainWriter implements the ChainWriterService interface.
+var _ ChainWriterService = (*chainWriter)(nil)
+
+func NewChainWriterService(config config.ChainWriter, logger logger.Logger, client evmclient.Client, txm evmtxmgr.TxManager) ChainWriterService {
+	return &chainWriter{logger: logger, client: client, config: config, txm: txm}
+}
+
+type chainWriter struct {
+	commonservices.StateMachine
+
+	logger logger.Logger
+	client evmclient.Client
+	config config.ChainWriter
+	txm    evmtxmgr.TxManager
+}
+
+func (w *chainWriter) SubmitSignedTransaction(ctx context.Context, payload []byte, signatures map[string]any, transactionID uuid.UUID, toAddress string, meta *types.TxMeta, value big.Int) error {
+	forwarderABI := evmtypes.MustGetABI(w.config.ABI())
+
+	// TODO(nickcorin):
+	// Check the required format for the signatures when packing the ABI. The original type was [][]byte, however we'll need strict type assertions
+	// translating any -> []byte.
+	//
+	// Also, double check that the method name is always going to be 'report'?
+	calldata, err := forwarderABI.Pack("report", common.HexToAddress(toAddress), payload, signatures)
+	if err != nil {
+		return fmt.Errorf("pack forwarder abi: %w", err)
+	}
+
+	// TODO(nickcorin): Change this to be config driven.
+	sendStrategy := txmgr.SendEveryStrategy{}
+
+	var checker evmtxmgr.TransmitCheckerSpec
+	if w.config.Checker() != "" {
+		checker.CheckerType = txmgrtypes.TransmitCheckerType(w.config.Checker())
+	}
+
+	req := evmtxmgr.TxRequest{
+		FromAddress:    w.config.FromAddress().Address(),
+		ToAddress:      w.config.ForwarderAddress().Address(),
```

Comment:
FromAddress/ToAddress should be a parameter that's passed in -- both of these values are being sourced from keystone/workflow specific config (that's incorrectly labeled as ChainWriter right now)

Reply:
Ah yeah this should be the `toAddress` passed into the method. The value that's now being treated as `toAddress` (the receiver address) will simply be a part of the input args:

```go
inputs := []any{receiverAddress, report, signatures}
CreateTransaction(ctx, inputs, workflowConfig.ForwarderAddress().Address(), ...)
```